src/executables/util_cs.c

```diff
@@ -3260,53 +3271,64 @@ applyinfo (UTIL_FUNCTION_ARG * arg)
 	  strcat (local_database_name, "@localhost");
 
 	  db_clear_host_connected ();
-
-	  if (check_database_name (local_database_name))
+	  if ((check_applied_info != false) && check_database_name (local_database_name))
 	    {
-	      goto check_applied_info_end;
+	      printf ("\n *** Applied Info. *** \n");
+	      check_applied_info = false;
+	      PRINT_AND_LOG_ERR_MSG ("\nERROR : %s\n", db_error_string (3));
 	    }
-	  if (db_login ("DBA", NULL) != NO_ERROR)
+	  if ((check_applied_info != false) && (db_login ("DBA", NULL) != NO_ERROR))
 	    {
-	      goto check_applied_info_end;
+	      printf ("\n *** Applied Info. *** \n");
+	      check_applied_info = false;
+	      PRINT_AND_LOG_ERR_MSG ("\nERROR : %s\n", db_error_string (3));
 	    }
-	  error = db_restart (arg->command_name, TRUE, local_database_name);
-	  if (error != NO_ERROR)
+	  if ((check_applied_info != false) && (db_restart (arg->command_name, TRUE, local_database_name) != NO_ERROR))
 	    {
-	      goto check_applied_info_end;
+	      printf ("\n *** Applied Info. *** \n");
+	      check_applied_info = false;
+	      PRINT_AND_LOG_ERR_MSG ("\nERROR : %s\n", db_error_string (3));
 	    }
 
-	  if (HA_DISABLED ())
+	  if ((check_applied_info != false) && HA_DISABLED ())
 	    {
+	      printf ("\n *** Applied Info. *** \n");
+	      check_applied_info = false;
+	      printf ("\nERROR : ");
 	      PRINT_AND_LOG_ERR_MSG (msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_APPLYINFO,
 						     APPLYINFO_MSG_NOT_HA_MODE));
-	      goto check_applied_info_end;
 	    }
+	  error = NO_ERROR;
 
-	  error = la_log_page_check (local_database_name, log_path, pageid, check_applied_info, check_copied_info,
-				     check_replica_info, verbose, &copied_eof_lsa, &copied_append_lsa,
+	  error = la_log_page_check (local_database_name, log_path, pageid, &check_applied_info, &check_copied_info,
+				     &check_replica_info, verbose, &copied_eof_lsa, &copied_append_lsa,
 				     &applied_final_lsa);
-	  (void) db_shutdown ();
+	  if (error == NO_ERROR)
```

Comment:
Why do you prevent the shutdown when with an error?

Reply:
Because of the printing error message.

Reply:
you mean `PRINT_AND_LOG_ERR_MSG` fails with this?

Reply:
Yes that's right.

Reply:
I think you must have tested it. In other utilities, however, it is used without `db_restart()`. For example, at the beginning of `compactdb()`, when an option is invalid they just PRINT_AND_LOG_ERR_MSG() and exit.

Reply:
When I consider `compactdb()`, there are using `er_init` and `er_final` both.

And `db_shutdown()` makes execute `er_final()`. So I must use print error message, before `db_shutdown()` is executed.

Reply:
But, it uses PRINT_AND_LOG_ERR_MSG () before er_init(). Another example is checksumdb(). Let me know the relation of PRINT_AND_LOG_ERR_MSG () and `er_` module.

Reply:
`PRINT_AND_LOG_ERR_MSG()` is used 2 ways.

1. PRINT_AND_LOG_ERR_MSG ("%s\n", db_error_string (3));

In this case, `er_set()` is necessary. So, `er_init()` is needed.

2. PRINT_AND_LOG_ERR_MSG (msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_APPLYINFO,						     APPLYINFO_MSG_NOT_HA_MODE));

In this case, it is not needed to use `er_set()`. 

But we use two ways together. To keep the message by `er_set()`, we should prevent db_shutdown till error message is printed.

Reply:
OK, the reason we have not to shutdown is to use db_error_string(), not PRINT_AND_LOG_ERR_MSG().